source/extensions/filters/http/cache/cache_headers_utils.cc

```diff
@@ -200,8 +200,18 @@ bool VaryHeader::hasVary(const Http::ResponseHeaderMap& headers) {
   return vary_header != nullptr && !vary_header->value().empty();
 }
 
+namespace {
+// The separator characters are used to create the vary-key, and must be characters that are
+// invalid to be inside values and header names. The chosen characters are invalid per:
+// https://tools.ietf.org/html/rfc2616#section-4.2.
+// Used to separate the values of different headers.
+const std::string header_separator = "\n";
```

Comment:
string constants with static lifetime should have type `constexpr std::string_view`. std::string has nontrivial construction and destruction, so the Envoy style guide forbids giving them static lifetime.

Reply:
I wasn't aware of that, thanks for pointing out.